### ISSUE TITLE
nuget.pack doesn't read input src files

### ISSUE THREAD

**Issue by JamieDixon** [mckn/gulp-nuget#7]
When running `nuget.pack` the src files aren't being taken into account.

The example given in the readme:

```
gulp.src(['./*.js', './package.json'])
        .pipe(nuget.pack({ nuspec: 'project.nuspec', nuget: nugetPath, version: '1.0.0' }))
        .pipe(gulp.dest('project.1.0.0.nupkg'));
```

implies that the files specified in `src` will be passed to `nuget.pack` and also that the output of `nuget.pack` (i.e. the package stream) will be passed to `gulp.dest`.

Perhaps I've misunderstood how this works but the result I get is quite different.

In my case, a package is created based on `project.nuspec` such that the output file *project.nupkg* is created during `nuget.pack` rather than by `gulp.dest`. The files specified in `gulp.src` are ignored and only files supplied in the `.nuspec` config are taken into account.

This means that to correctly create a package I need to specify my files in my `project.nuspec` and do the following:

```
gulp.src('')
.pipe(nuget.pack({ nuspec: 'project.nuspec', nuget: nugetPath, version: '1.0.0' }))
```

This generates the package correctly but the whole thing feels a bit odd. Like I say, I'm probably missing something obvious.

**Comment by NightOwl888** [comment]
I agree this is odd. But I don't agree that the solution should be to make the input source files, since that is already a feature of NuGet (using the `<files>` section or a directory convention).

There should be a way to run the command on multiple `.nuspec` files rather than including a single one in the input. Something like.

```
gulp.src('./*.nuspec')
    .pipe(nuget.pack({ nuget: './nuget.exe', version: '1.0.0' }))
    .pipe(out(distributionFolder + '{basename}.nupkg'));
```

Currently, if I do this I get an error `".nuspec file path missing from options."`

So far this is the only way I have found to make it find and run all of my `.nuspec` files. The script uses [gulp-out](https://www.npmjs.com/package/gulp-out) and [glob](https://www.npmjs.com/package/glob).

```
    // Get the nuspec files
    var nuspecFiles = glob.sync("./**/*.nuspec");

    console.log('Nuspec files: ' + nuspecFiles);

    var nuspecLength = nuspecFiles.length;
    for (var i = 0; i < nuspecLength; i++) {
        var nuspecFile = nuspecFiles[i];
        
        gulp.src('')
            .pipe(nuget.pack({ nuspec: nuspecFile, nuget: nugetPath, version: version }))
            .pipe(out(distributionFolder + '{basename}.nupkg'));
    }
```

Of course, using this approach there doesn't appear to be any reasonable way to make the task synchronous without resorting to `setTimeout()`.

**Comment by mckn** [comment]
Should be resolved in new version 1.0.0
